cpp/open3d/core/Tensor.cpp

```diff
@@ -371,6 +371,15 @@ Tensor& Tensor::operator=(Tensor&& other) && {
     return *this;
 }
 
+Tensor Tensor::ReinterpretCast(const core::Dtype& dtype) const {
+    if (dtype_.ByteSize() != dtype.ByteSize()) {
+        utility::LogError(
+                "Cannot reinterpret cast between data-types of different "
```

Comment:
Print the name and byte size of the dtypes in the error message.

Reply:
Done.